alcli/alertlogic_cli.py

```diff
@@ -249,13 +249,24 @@ def _encode(self, operation, param_name, param_value):
             p = urlparse(param_value)
             if p.scheme == "file":
                 value_file_path = os.path.abspath(os.path.join(p.netloc, p.path))
-                with open (value_file_path, "r") as value_file:
+                with open(value_file_path, "r") as value_file:
                     param_value = value_file.read()
 
         schema = operation.get_schema()
         parameter = schema[OpenAPIKeyWord.PARAMETERS][param_name]
+        type = parameter.get(OpenAPIKeyWord.TYPE)
+        oneof = parameter.get(OpenAPIKeyWord.ONE_OF)
+        anyof = parameter.get(OpenAPIKeyWord.ANY_OF)
+        allof = parameter.get(OpenAPIKeyWord.ALL_OF)
+
+        if oneof or anyof:
+            # Take first type available, assuming same parameter must not be of different type
```

Comment:
Can we cause an internal error on schema load or in the validation pipeline if this isn't the case? I'd like to avoid silent problems here.

Reply:
Yes, we can include check which will check or all types to be equal for oneof and anyof and for allof to have at least in one of the pieces.

Reply:
(in the validation routine)